doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,27 @@
+.. program:: ghdl
+.. _COSIM:VHPIDIRECT:Examples:arrays:
+
+Arrays
+#######
+
+:cosimtree:`A Bounded Array Sized in C <vhpidirect/arrays/intvector>`
+*********************************************************************
+
+Integer arrays fully defined in C can be passed to VHDL by first passing their size, so that an appropriate array 
+type can be created to hold the array's pointer. After that, a VHDL subprogram can be defined to return the array access.
+
+:cosimtree:`A Bounded Array Sized in C: First defined in main() <vhpidirect/arrays/intvector/maindefined>`
+----------------------------------------------------------------------------------------------------------
+
+If the integer array must be created or filled by some more advanced process before VHDL can request the array's pointer, it is
+possible to execute the GHDL simulation within a custom ``int main()`` entrypoint (see :ref:`COSIM:VHPIDIRECT:Examples:wrapping:basic`).
+In this example the custom entrypoint prompts the user for the size of the array and then handles the variables appropriately before
+going on to execute the GHDL simulation.
+
+:cosimtree:`A Bounded Array Sized in VHDL <vhpidirect/arrays/intvectorgeneric>`
```

Comment:
I think it is better to keep this as a subexample of `intvector`. If you want, just move the current content of `intvector` to say `intvector/sizedinc` and rename this one to `intvector/sizedinvhdl`.

The main point of this (intvectorgeneric) is NOT the usage of generics, but the fact that the size is provided from VHDL. Hence, the introduction of the subexample should not focus on generics.

We can keep the description of the generics in a HINT. However, it is not required, because generics can have defaults. Moreover, it might make sense to move use a constant instead, and move the usage of generics to a different `quickstart/cli` example. This is because it is also possible to manipulate `argc` and `argv` before calling `ghdl_main`. Hence, it is feasible to have some size defined by a generic in VHDL, but have it defined in C in practice, by passing it as a `-gGENERIC=` to `ghdl_main`. If such an entrypoint is combined with the current `intvectorgeneric` example, then VHDL will pass that arg back to C when calling `getIntArr_ptr`. Although this might feel as a rare use case, it is easy for it to happen when adapting a design that is not aware of VHPIDIRECT features, and we want to optionally enhance it with them. Which is, precisely, the foundation of the non-intrusive philosophy of VUnit (and, hence, VUnit/cosim).